jupyter-extension/jupyterlab_pachyderm/pps_client.py

```diff
@@ -165,7 +198,7 @@ def create_pipeline_spec(config: PpsConfig, companion_branch: str) -> dict:
     companion_repo = f"{config.pipeline.name}__context"
     input_spec = dict(
         cross=[
-            config.input_spec,
+            yaml.safe_load(config.input_spec),
```

Comment:
rather than doing it here, I would do it within the `PpsConfig.from_notebook` method. That way this method shouldn't reasonably error and you can catch an invalid yaml on parsing.